tests/integration/test_traefik_ingress.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+"""Temporal Server charm integration with Traefik tests."""
+
+import asyncio
+import json
+import logging
+from pathlib import Path
+
+import grpc
+import pytest
+import pytest_asyncio
+import yaml
+from grpc_health.v1 import health_pb2, health_pb2_grpc
+from ops import Application
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+TEMPORAL_ADMIN = "temporal-admin-k8s"
+TEMPORAL_ADMIN_CHANNEL = "stable"
+POSTGRESQL_K8S = "postgresql-k8s"
+POSTGRESQL_K8S_CHANNEL = "14"
+POSTGRESQL_K8S_TRUST = True
+TRAEFIK_K8S = "traefik-k8s"
+TRAEFIK_K8S_CHANNEL = "latest/stable"
+TRAEFIK_K8S_TRUST = True
+
+
+@pytest_asyncio.fixture(name="deploy", scope="module")
+async def deploy(ops_test: OpsTest):
```

Comment:
kind of not absolutely necessary since the test has already been implemented: but it would be nice to use jubilant for this test if the effort to refactor is not substantial. from what i remember, jubilant is quite intuitive and designed to be used in a similar way to the juju cli (so it may not be too intense to refactor)

happy even if we use ops_test for this test - we can consider using jubilant later when we add more tests